instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -56,7 +56,9 @@ def on_start(request, _)
 
             parent_context = extract_remote_context(request)
             span = create_span(parent_context, request)
-            request.env[TOKENS_KEY] = register_current_span(span)
+            span_ctx = OpenTelemetry::Trace.context_with_span(span, parent_context: parent_context)
+            rack_ctx = OpenTelemetry::Instrumentation::Rack.context_with_span(span, parent_context: span_ctx)
+            request.env[TOKEN_KEY] = OpenTelemetry::Context.attach(rack_ctx)
```

Comment:
We don't need to attach multiple context objects just a single context that captures:
- The extracted context (this is where we get our baggage)
- The current span context
- The current rack span context

The `*.context_with_span` calls make use of the [set_value](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/api/lib/opentelemetry/context.rb#L148-L157) context interface which always returns a new ctx object.

So instead of layering contexts, we're building a single one here that contains all the entries needed when we start a rack request.

Reply:
TIL